File: 백준/Silver/10866. 덱/덱.py
from _collections import deque
import sys

q = deque()

for _ in range(int(sys.stdin.readline())):
    li = list(sys.stdin.readline().split())

    if li[0] == 'push_front':
        q.appendleft(li[1])

    elif li[0] == 'push_back':
        q.append(li[1])

    elif li[0] == 'pop_front':
        if q:
            print(q.popleft())
        else:
            print(-1)

    elif li[0] == 'pop_back':
        if q:
            print(q.pop())
        else:
            print(-1)

    elif li[0] == 'size':
        print(len(q))

    elif li[0] == 'empty':
        if q:
            print(0)
        else:
            print(1)

    elif li[0] == 'front':
        if q:
            print(q[0])
        else:
            print(-1)

    elif li[0] == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)

File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        ArrayDeque<Integer> arr = new ArrayDeque<>();

        while(N!=0) {
            String[] li = br.readLine().split(" ");
            switch (li[0]) {
                case "push_front":
                    arr.addFirst(Integer.parseInt(li[1]));
                    break;
                case "push_back":
                    arr.addLast(Integer.parseInt(li[1]));
                    break;
                case "pop_front":
                    bw.write((arr.isEmpty()?-1:arr.removeFirst())+"\n");
                    break;
                case "pop_back":
                    bw.write((arr.isEmpty()?-1:arr.removeLast())+"\n");
                    break;
                case "size":
                    bw.write(arr.size()+"\n");
                    break;
                case "empty":
                    bw.write((arr.isEmpty()?1:0)+"\n");
                    break;
                case "front":
                    bw.write((arr.isEmpty()?-1:arr.getFirst())+"\n");
                    break;
                case "back":
                    bw.write((arr.isEmpty()?-1:arr.getLast())+"\n");
                    break;
            }
            N--;
        }
        bw.flush();
        bw.close();
    }
}
